src/core/ActionForInterface.h

```diff
@@ -0,0 +1,89 @@
+/* +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+   Copyright (c) 2017-2023 The plumed team
+   (see the PEOPLE file at the root of the distribution for a list of names)
+
+   See http://www.plumed.org for more information.
+
+   This file is part of plumed, version 2.
+
+   plumed is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   plumed is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with plumed.  If not, see <http://www.gnu.org/licenses/>.
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ */
+#ifndef __PLUMED_core_ActionForInterface_h
+#define __PLUMED_core_ActionForInterface_h
+
+#include "tools/TypesafePtr.h"
+#include "ActionWithValue.h"
+
+namespace PLMD {
+
+class ActionForInterface : public ActionWithValue {
+  friend class Atoms;
+  friend class PlumedMain;
+protected:
+// Is this the first step
+  bool firststep;
+/// Have the forces in this action been scaled by another action
+  bool wasscaled;
+/// Action has been set
+  bool wasset;
+/// The role this plays
+  std::string role;
```

Comment:
If I understood where this is going, I think `role` will benefit, a lot, by being converted to be a `enum class` mainly because will be faster (I saw it used a lot in `if(a->getRole()=="something")` with something being only x,y,z,q, or m) and occupy less space, moreover it will be self documenting and with being an `enum class` also more able to catch up some bugs at compile time.

Reply:
Daniele, the 6 calls in the whole code that have the form `if(a->getRole()=="something")` appear in constructors.  These calls are NOT slowing down the code.